### PR TITLE
feat: update docs for celestia-node v0.12.1

### LINKED ISSUE #1308: Update docs for node
* New release https://github.com/celestiaorg/celestia-node/releases/tag/v0.12.1
* New build option `make build-jemalloc` (details on the release)
* Decreased resource requirements
	* BN/FN -> to 4GiB min RAM
	* LN -> 500MiB min RAM

### PR DESCRIPTION
**Pull request by jcstein**
## Overview

Updates docs for v0.12.1 celestia-node, along with new build command and updated memory requirements. Resolves #1308 

- [x] update memory requirements
- [x] add new build command
- [x] update graphic

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

### CODE REVIEW

**Review by YazzyYaz — APPROVED** [review]

File: developers/node-tutorial.md [hunk 1]
```diff
@@ -237,9 +237,13 @@ commands:
 3. Build the `celestia` binary:
 
    ```bash
-   make build
+   make build-jemalloc
```
Comment (Wondertan) [inline developers/node-tutorial.md:240]: We should keep the regular one and add jemalloc as a secondary option.

File: developers/node-tutorial.md [hunk 2]
```diff
@@ -237,9 +237,13 @@ commands:
...
    ```
 
+   ::: tip NOTE
+   This build option enables CGO. [Learn more about the build command](https://github.com/celestiaorg/celestia-node/releases/tag/v0.12.1#:~:text=%F0%9F%8F%97%EF%B8%8F-,New%20build%20option,-%3A%20Makefile%20now%20has).
```
Comment (Wondertan) [inline developers/node-tutorial.md:244]: Another thing I just realized is that this build downloads and installs jemalloc. It doesn't hurt to mention that.

File: developers/node-tutorial.md (hunk 1 above)
Comment (jcstein) [reply developers/node-tutorial.md:240]: sounds good, sorry, was going off of [this](https://github.com/celestiaorg/docs/issues/1308#issuecomment-1852434735)

File: developers/node-tutorial.md (hunk 2 above)
Comment (jcstein) [reply developers/node-tutorial.md:244]: got it! following up in PR

File: developers/node-tutorial.md (hunk 1 above)
Comment (Wondertan) [reply developers/node-tutorial.md:240]: Miscomunice. I thought of this as a secondary option for all the node types

Comment (jcstein) [reply developers/node-tutorial.md:240]: we good! just made PR :smile:

**Pull request merged**